src/mission-control/infra/database/postgresjs/repositories/core-domain.repositorie.ts

```diff
@@ -1,27 +1,90 @@
 import { Injectable } from '@nestjs/common'
 
-import { CoreDomainRepository } from 'src/mission-control/domain/tasks/repositories/core-domain-repository'
+import { ConfidenceTag } from '@adapters/confidence-tag/confidence-tag.enum'
+import {
+  CoreDomainRepository,
+  findKeyResultByConfidenceInput,
+  findOutdatedKeyResultInput,
+} from 'src/mission-control/domain/tasks/repositories/core-domain-repository'
 
 import { PostgresJsService } from '../postgresjs.service'
 
 import {
   CoreDomainDataMapper,
+  keyResultFromMCContextOutputTable,
   userFromMCContextOutputTable,
 } from './data-mappers/core-domain.data-mapper'
 
 @Injectable()
 export class PostgresJsCoreDomainRepository implements CoreDomainRepository {
   constructor(private readonly postgres: PostgresJsService) {}
+
+  async findOneKeyResultWithOutdatedCheckin({ ownerId, teamId }: findOutdatedKeyResultInput) {
+    const [queryOutput] = await this.postgres.getSqlInstance()<keyResultFromMCContextOutputTable[]>`
+    SELECT kr.id AS keyresult_id
```

Comment:
Aqui vale o mesmo que comentei lá em cima: acho que dá pra simplificar a query (e consequentemente deixá-la mais rápida) usando uma combinação de group, order e limit

Reply:
Fiz um commit nessa branch com algumas alterações em `src/mission-control/infra/database/postgresjs/repositories/core-domain.repositorie.ts`. Pode dar um confere, pls?

Reply:
Consegue me passar o link da branch? Eu to vendo uma versão que não sei se é a que vc tá se referindo